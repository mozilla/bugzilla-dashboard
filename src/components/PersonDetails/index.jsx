import React from 'react';
import PropTypes from 'prop-types';
import DetailView from '../DetailView';
import BugzillaComponents from '../BugzillaComponents';

class PersonDetails extends React.Component {
  render() {
    const { person, bugzillaComponents, onGoBack } = this.props;

    const components = Object.values(bugzillaComponents)
      .reduce((result, bzComponent) => {
        const { bugzillaEmail, product, component } = bzComponent;
        if (bugzillaEmail === person.bugzillaEmail) {
          // eslint-disable-next-line no-param-reassign
          result[`${product}::${component}`] = bzComponent;
        }
        return result;
      }, {});

    return (
      <DetailView title={person.cn} onGoBack={onGoBack}>
        <BugzillaComponents
          bugzillaComponents={components}
        />
      </DetailView>
    );
  }
}

PersonDetails.propTypes = {
  person: PropTypes.shape({}).isRequired,
  bugzillaComponents: PropTypes.shape({}).isRequired,
  onGoBack: PropTypes.func.isRequired,
};

export default PersonDetails;
