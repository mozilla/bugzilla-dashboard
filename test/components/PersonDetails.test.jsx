import React from 'react';
import renderer from 'react-test-renderer';
import PersonDetails from '../../src/components/PersonDetails';
import partialOrg from '../mocks/partialOrg';
import bugzillaComponents from '../mocks/bugzillaComponents';

it('renders the details for an individual contributor', () => {
  const person = partialOrg['someone@mozilla.com'];
  const tree = renderer
    .create((
      <PersonDetails
        person={person}
        bugzillaComponents={bugzillaComponents}
        onGoBack={() => null}
      />
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the details for a manager', () => {
  const person = partialOrg['manager@mozilla.com'];
  const tree = renderer
    .create((
      <PersonDetails
        person={person}
        bugzillaComponents={bugzillaComponents}
        onGoBack={() => null}
      />
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
