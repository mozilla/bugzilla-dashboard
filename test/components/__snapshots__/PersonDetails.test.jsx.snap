// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the details for a manager 1`] = `
<div
  className="DetailView-root-1"
>
  <div>
    <a
      href="/"
      onClick={[Function]}
      rel="noopener noreferrer"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-3"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </a>
  </div>
  <div>
    <h2
      className="DetailView-title-2"
    >
      Manager
    </h2>
  </div>
</div>
`;

exports[`renders the details for an individual contributor 1`] = `
<div
  className="DetailView-root-1"
>
  <div>
    <a
      href="/"
      onClick={[Function]}
      rel="noopener noreferrer"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-3"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </a>
  </div>
  <div>
    <h2
      className="DetailView-title-2"
    >
      Someone
    </h2>
  </div>
</div>
`;
